src/context.nim

```diff
@@ -6,293 +6,79 @@
 #    distribution, for details about the copyright.
 #
 
-import std / [strutils, os, tables, sets, json,
-  terminal, hashes, uri]
-import versions, parse_requires, compiledpatterns, osutils
-
-export tables, sets, json
-export versions, parse_requires, compiledpatterns
+import std / [os]
+import versions, parse_requires, compiledpatterns, reporters
 
 const
-  MockupRun* = defined(atlasTests)
   UnitTests* = defined(atlasUnitTests)
   TestsDir* = "atlas/tests"
 
 const
   AtlasWorkspace* = "atlas.workspace"
 
-proc getUrl*(input: string): PackageUrl =
-  try:
-    var input = input
-    input.removeSuffix(".git")
-    let u = PackageUrl(parseUri(input))
-    if u.scheme in ["git", "https", "http", "hg", "file"]:
-      result = u
-  except UriParseError:
-    discard
-
-export uri.`$`, uri.`/`, uri.UriParseError
-
 type
   CfgPath* = distinct string # put into a config `--path:"../x"`
 
   SemVerField* = enum
     major, minor, patch
 
-  ResolutionAlgorithm* = enum
-    MinVer, SemVer, MaxVer
-
   CloneStatus* = enum
     Ok, NotFound, OtherError
 
-  PackageName* = distinct string
-  PackageDir* = distinct string
-  PackageNimble* = distinct string
-  PackageRepo* = distinct string
```

Comment:
Dropping these back to just strings will lead back to the state where all of these types are mixed up in the code. That caused a number of bugs/issues where package dir was used instead of package name, etc.

Reply:
The distinction between `PackageName` and `PackageDir` was not clear. Also the new depedency graph is more flexible and allows for arbitrary packages to be added to it regardless of where on the filesystem it is.